libs/partners/openai/langchain_openai/chat_models/base.py

```diff
@@ -317,10 +317,26 @@ class BaseChatOpenAI(BaseChatModel):
         None."""
     max_retries: int = 2
     """Maximum number of retries to make when generating."""
+    presence_penalty: float = 0
+    """Penalizes repeated tokens."""
+    frequency_penalty: float = 0
+    """Penalizes repeated tokens according to frequency."""
```

Comment:
sorry another one. Would it make sense to make these optional, and not pass them when they're None? Then we rely on any defaults openai sets.